source/SpinalHDL/Semantic/when_switch.rst

```diff
@@ -97,6 +97,37 @@ is equivalent to
     }
   }
 
+
+Additional options
+^^^^^^^^^^^^^^^^^^
+
+By default, SpinalHDL will generate an "UNREACHABLE DEFAULT STATEMENT" error if a switch/is contains a ``default`` statement while already all the possible logical values of the 'switch' are covered by the 'is' definitions. You can drop this error reporting by specifying `` switch(myValue, coverUnreachable = true){ ... }``.
+
+.. code-block:: scala
+  
+  switch(my2Bits, coverUnreachable = false) {
+      is(0) { ... }
+      is(1) { ... } 
+      is(2) { ... }
+      is(3) { ... }
+      default { ... } //This will be ok
+  }
+  
+Note, this check is done on the logical values, not on the physical values. For instance, if you have a SpinalEnum(A,B,C) encoded in a on-hot manner, SpinalHDL will only care about the A,B,C values ("001" "010" "100"). Pyhsical values as "000" "011" "101" "110" "111" will not be taken in account.
+
+
+By default, SpinalHDL will generate an "DUPLICATED ELEMENTS IN SWITCH IS(...) STATEMENT" error if a given ``is`` statements provides multiple time the same value. For instance ``is(42,42) { .. }`` 
+You can drop this error reporting by specifying `` switch(myValue, strict = true){ ... }``. SpinalHDL will then take care of removing duplicated values.
+
+.. code-block:: scala
+  
+  switch(value, strict = false) {
+      is(0)  { ... }
+      is(1,1,1,1,1)  { ... } //This will be ok
+      is(2) { ... }
+  }
```

Comment:
```suggestion
  switch(value, strict = false) {
      is(0) { ... }
      is(1,1,1,1,1) { ... } // This will be okay
      is(2) { ... }
  }
```